modules/mtl/arrow-mtl-data/src/test/kotlin/arrow/mtl/WriterTTest.kt

```diff
@@ -121,3 +118,13 @@ class WriterTTest : UnitSpec() {
     )
   }
 }
+
+private fun <F, W> WriterT.Companion.genK(
+  genkF: GenK<F>,
+  genW: Gen<W>
+) = object : GenK<WriterTPartialOf<F, W>> {
+  override fun <A> genK(gen: Gen<A>): Gen<Kind<WriterTPartialOf<F, W>, A>> =
+    genkF.genK(Gen.tuple2(genW, gen)).map {
+      WriterT(it)
```

Comment:
This one is nice ^-^ It generates output and result, no longer do we test with a writer that is isomorphic to id:
`Writer<W, A> = () -> Tuple2<W, A> = Tuple2<W, A>` and if `W` is `Nothing` which is the case when using `Applicative.just()` it is equal to `A`. But now since `W` is no longer `Nothing` we finally get good tests :)

Also you can shorten those a bit by using `map(::WriterT)` ^^